Netkan/Sources/Curse/CurseFile.cs

```diff
@@ -30,22 +22,24 @@ public class CurseFile
         /// <summary>
         /// Returns the direct path to the file
         /// </summary>
-        /// <returns>The download url</returns>
-        public String GetDownloadUrl()
+        /// <returns>
+        /// The download URL
+        /// </returns>
+        public string GetDownloadUrl()
         {
-            if (_downloadUrl == null)
+            if (string.IsNullOrEmpty(_downloadUrl))
             {
-                _downloadUrl = CurseApi.ResolveRedirect(new Uri(Mod.GetPageUrl() + "/files/" + id + "/download")).ToString();
```

Comment:
This also still works on my test.

Reply:
I'll change it back. I think I was trying to rewrite it to rely only on properties available in the new API's JSON, but if the old stuff still works then we might as well still use it.